includes/class-wc-checkout.php

```diff
@@ -744,6 +746,13 @@ protected function validate_posted_data( &$data, &$errors ) {
 				$format      = array_filter( isset( $field['validate'] ) ? (array) $field['validate'] : array() );
 				$field_label = isset( $field['label'] ) ? $field['label'] : '';
 
+				if ( 'country' === $field['type'] &&
+					! ( 'shipping' === $fieldset_key && ! $data ['ship_to_different_address'] ) &&
```

Comment:
should we add a clause for `$required` and `isset( $data[ $key ] )` incase country is not a required field at all. based on the code in callee functions, seems like if this field is not present, we can use `WC()->customer->get_shipping_country()` as a backup.

Reply:
But this code is part of the loop that iterates over the posted data, so it will validate the posted country only if there is any. Also there's a `if ( ! isset( $data[ $key ] ) ) continue;` at the beginning of the loop. Or you mean something else?

The `! ( 'shipping' === $fieldset_key && ! $data ['ship_to_different_address'] )` is there to prevent a duplicate "Invalid country code" error, since when "ship to different address" is unchecked then shipping and billing countries will be the same ones.

Reply:
> But this code is part of the loop that iterates over the posted data

Ah sorry, I meant `empty` not `isset`.  Basically, if a field is not required then it can be empty as well right? Or not? We should probably bail from checking if the country is valid, if its not a required field and is empty. 

> The ! ( 'shipping' === $fieldset_key && ! $data ['ship_to_different_address'] ) is there to prevent a duplicate "Invalid country code" error, since when "ship to different address" is unchecked then shipping and billing countries will be the same ones.

Makes sense :+1:

Reply:
Ok, a couple of things:

1. Theres's a `$validate_fieldset` variable already for that, generated by calling `maybe_skip_fieldset`; so we can just use that.
2. `maybe_skip_fieldset` is already doing the `'shipping' === $fieldset_key && ! $data ['ship_to_different_address'] ` part so there's no need to repeat it.

So I've pushed a commit with these changes.